include/checker/dd/TaskManager.hpp

```diff
@@ -0,0 +1,127 @@
+/*
+* This file is part of MQT QCEC library which is released under the MIT license.
+* See file README.md or go to http://iic.jku.at/eda/research/quantum_verification/ for more information.
+*/
+
+#pragma once
+
+#include "QuantumComputation.hpp"
+#include "dd/Package.hpp"
+
+namespace ec {
+    enum Direction : bool { Left  = true,
+                            Right = false };
+
+    template<class DDType>
+    class TaskManager {
+    protected:
+        const qc::QuantumComputation* qc = nullptr;
+        qc::Permutation               permutation{};
+        decltype(qc->begin())         iterator;
+        decltype(qc->end())           end;
+        std::unique_ptr<dd::Package>& package;
+        DDType                        internalState{};
+        ec::Direction                 direction = Left;
+
+    public:
+        explicit TaskManager(const qc::QuantumComputation& qc, std::unique_ptr<dd::Package>& package, const ec::Direction& direction = Left) noexcept:
+            qc(&qc), package(package), direction(direction) {
+            permutation = qc.initialLayout;
+            iterator    = qc.begin();
+            end         = qc.end();
+        }
+
+        [[nodiscard]] bool finished() const noexcept { return iterator == end; }
+
+        const std::unique_ptr<qc::Operation>& operator()() const { return *iterator; }
+
+        [[nodiscard]] const DDType& getInternalState() const noexcept {
+            return internalState;
+        }
+        void setInternalState(const DDType& state) noexcept {
+            internalState = state;
+        }
+        void flipDirection() noexcept { direction = (direction == Left) ? Right : Left; }
+
+        inline qc::MatrixDD getDD() { return (*iterator)->getDD(package, permutation); }
+        inline qc::MatrixDD getInverseDD() { return (*iterator)->getInverseDD(package, permutation); }
```

Comment:
Definitely. I guess Clion just didn't propose it because these functions are not marked `const` (and actually cannot be marked const due to the `getDD` functions not being `const`).